vllm/worker/hpu_model_runner.py

```diff
@@ -724,15 +735,17 @@ def load_model(self) -> None:
 
             hidden_layer_markstep_interval = int(
                 os.getenv('VLLM_CONFIG_HIDDEN_LAYERS', '1'))
+            model_config = getattr(self.model, "config", None)
             modify_decoder_layer(
```

Comment:
What about encoder layers? I believe we should also add mark_step after each encoder layer. It should have two benefits:

1. Lower memory usage.
2. Better performance - currently we need to accumulate entire EncoderModel to start execution on device (about 35ms in common scenario). If we will break graph after each encoder layer, then we can start execution much faster.

Reply:
I think the encoder layers don’t need to be wrapped in mark_step. Here are the reasons:
1. The encoder layers are part of the vision model component in Mllama, which is currently not wrapped in the HPU graph. This is because certain operations in the vision model fall back to the CPU, leading to graph-building failures. At present, only the language model component is wrapped in the HPU graph.
2. This may have small impact on performance, because most of time is spent in the decoding phase, where only the language model component is executed.